core/src/main/scala/org/http4s/rho/ExecutableCompiler.scala

```diff
@@ -17,57 +15,16 @@ trait ExecutableCompiler {
   //////////////////////// Stuff for executing the route //////////////////////////////////////
 
   /** Walks the validation tree */
-  def ensureValidHeaders(v: HeaderRule, req: Request): ResultResponse[HList] =
-    runValidation(req, v, HNil)
+  def runRequestRules(v: RequestRule, req: Request): ResultResponse[HList] =
+    runRequestRules(req, v, HNil)
 
-  /** The untyped guts of ensureValidHeaders and friends */
-  def runValidation(req: Request, v: HeaderRule, stack: HList): ResultResponse[HList] = {
-    import bits.HeaderAST.MetaCons
-    v match {
-      case HeaderAnd(a, b) => runValidation(req, a, stack).flatMap(runValidation(req, b, _))
 
-      case HeaderOr(a, b) => runValidation(req, a, stack).orElse(runValidation(req, b, stack))
-
-      case HeaderExists(key, f) => req.headers.get(key) match {
-        case Some(h) => 
-          scalaz.\/.fromTryCatchNonFatal( f(h) ) match {
-            case scalaz.-\/(t) => FailureResponse.badRequest(t.getMessage())
```

Comment:
This security leak was changed to a `InternalServiceError` status with a generic message while logging the error. See https://github.com/http4s/rho/pull/115/files#diff-7f2fd9ea04632bd23bdfc7136dcb1b37R156